src/main/java/org/libsodium/jni/crypto/SealedBox.java

```diff
@@ -0,0 +1,56 @@
+package org.libsodium.jni.crypto;
+
+import org.libsodium.jni.encoders.Encoder;
+
+import static org.libsodium.jni.NaCl.sodium;
+import static org.libsodium.jni.SodiumConstants.SEAL_BYTES;
+import static org.libsodium.jni.crypto.Util.isValid;
+
+public class SealedBox {
+
+    private byte[] publicKey;
+    private byte[] privateKey;
+
+    public SealedBox(byte[] publicKey) {
+        if (publicKey == null) {
+            throw new IllegalArgumentException("Public key must not be null");
+        }
+        this.publicKey = publicKey;
+        this.privateKey = null;
+    }
+
+    public SealedBox(String publicKey, Encoder encoder) {
+        this(encoder.decode(publicKey));
+    }
+
+    public SealedBox(byte[] publicKey, byte[] privateKey) {
+        if (publicKey == null) {
+            throw new IllegalArgumentException("Public key must not be null");
+        }
+        if (privateKey == null) {
+            throw new IllegalArgumentException("Private key must not be null");
+        }
+        this.publicKey = publicKey;
+        this.privateKey = privateKey;
+    }
+
+    public SealedBox(String publicKey, String privateKey, Encoder encoder) {
+        this(encoder.decode(publicKey), encoder.decode(privateKey));
+    }
+
+    public byte[] encrypt(byte[] message) {
+        byte[] ct = new byte[message.length + SEAL_BYTES];
+        isValid(sodium().crypto_box_seal(
```

Comment:
The Util class is marked deprecated and should not be used.

isValid is supposed to return a boolean, though throws a RuntimeException.

Reply:
Ok, I have removed the usage of isValid and have moved the check internally.

What's the planned approach to deal with all the deprecated classes ?

Reply:
Planning to remove the deprecated classes in a major release.
The deprecated classes haven't been removed yet as it's unclear impact to downstream developers.